tests/DCPS/TcpReconnect/publisher.cpp

```diff
@@ -89,7 +89,7 @@ int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
       ACE_TString command_line = stubCmd + ACE_TEXT(" ") + stubArgs;
       ACE_DEBUG((LM_INFO, ACE_TEXT("stub command line: %s\n"), command_line.c_str()));
 
-      if (options.command_line(ACE_TEXT_CHAR_TO_TCHAR(command_line.c_str())) != 0)
+      if (options.command_line(ACE_TEXT(command_line.c_str())) != 0)
```

Comment:
This isn't going to compile.

Reply:
On wchar builds that is.

Reply:
To satisfy the warning, the format string needs to be a string literal:

```suggestion
      if (options.command_line("%s", ACE_TEXT_ALWAYS_CHAR(command_line.c_str())) != 0)
```

Reply:
I pushed a new fix before I saw your comments. Do you think this version would work too?

Reply:
Instead of converting the command_line string to plain ascii you can also use the correct format string, based on wchar or not, use `%s` or `%ls`, see for example https://github.com/DOCGroup/ACE_TAO/blob/368d8bff6c4b9889eb67d6337b3ebfaf5a6bca2c/ACE/tests/Process_Manager_Test.cpp#L84

Reply:
The version I wrote is based on similar code in TAO_IDL be_interface.cpp and utl_global.cpp. I welcome and appreciate other  suggestions. Now, when we have more than one, which one would be the best?

Reply:
> Now, when we have more than one, which one would be the best?

More than one of what?

Reply:
"suggestions". We have three options now (unless the one in the PR now is wrong - then we have two).

Reply:
Personally I'd prefer the one that minimizes the involvement of `ACE_USES_WCHAR` because I don't like anything about the whole system, since it leads to confusion like this. However what Johnny suggested is the "correct" way to use the system since it allows for the possibility of wide characters in `command_line`. It's up to you to choose.